### PR TITLE
gtime: Make ParseInterval deterministic

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Fix gtime.ParseInterval so it's deterministic, i.e. base it on UTC so daylight saving doesn't cause surprises.

**Which issue(s) this PR fixes**:
Backend tests are currently failing due to a discrepancy in interval calculation caused by daylight saving.

**Special notes for your reviewer**:
@marefr does this make sense, or should we fix the failing tests instead?

### CODE REVIEW

**Review by dsotirakis — COMMENTED** [review]
Looks good, although would be good to check if we can replace / merge with 

https://github.com/grafana/grafana/blob/master/pkg/components/gtime/gtime.go#L42

?

They are doing more or less the same thing?

**Review by marefr — APPROVED** [review]
LGTM

File: pkg/components/gtime/gtime.go [hunk 1]
```diff
@@ -22,16 +22,18 @@ func ParseInterval(inp string) (time.Duration, error) {
...
 	num := int(dur)
 
-	now := time.Now()
+	// Use UTC to ensure that the interval is deterministic, and daylight saving
+	// doesn't cause surprises
+	now := time.Now().UTC()
 	switch period {
 	case "d":
-		return now.Sub(now.AddDate(0, 0, -num)), nil
+		return now.AddDate(0, 0, num).Sub(now), nil
```
Comment (marefr) [inline pkg/components/gtime/gtime.go:30]: To get rid of relative to the current wall time, can we return `dur * 24 * time.Hour` and so on?

Comment (aknuds1) [reply pkg/components/gtime/gtime.go:30]: @marefr this function is meant to be relative to wall time, if we want fixed durations, `ParseDuration` should be used instead. I'm actually wondering if we can use fixed durations everywhere? Would certainly make things easier.

**Pull request merged**